### ISSUE TITLE
In definition of 'keyboard interface' and in 2.1.1, "would satisfy the success criterion" is incomplete

### ISSUE THREAD

**Issue by mitchellevan** [w3c/wcag2ict#421]
### Currently

The following note appears in two places:

- [Applying SC 2.1.1 Keyboard to Non-Web Documents and Software](https://www.w3.org/TR/wcag2ict-22/#applying-sc-2-1-1-keyboard-to-non-web-documents-and-software)
- [definition of 'keyboard interface'](https://www.w3.org/TR/wcag2ict-22/#dfn-keyboard-interface) 

> Keyboard interface does not refer to a physical device but to the interface between the software and any keyboard or keyboard substitute (i.e., an interface where the software accepts text or other keystroke input). Underlying [platform software](#platform-software) may provide device independent input services to applications that enable operation via such a keyboard interface. [Software](#software) that supports operation via such platform device independent services would be operable via a keyboard interface and would satisfy the success criterion.

There are issues in the last sentence.

(1) Issue in context of 2.1.1:

* The sentence is not correct, because supporting an input service is necessary to meet 2.1.1, but not sufficient. Not just the software program as a whole, but "all functionality of the content" must be "operable through a keyboard interface without requiring specific timings for individual keystrokes" (except for essential path-based input). It's okay for a note to address just part of a success criterion, but as currently written this note could be misunderstood as reinterpreting the SC to be less stringent for software.

(2) Issue in context of the glossary:

* Since we're in the glossary, it's unclear which "success criterion" the note is referring to. The term 'keyboard interface' is used directly in two Level A criteria: SC 2.1.1 Keyboard and SC 2.1.2 No Keyboard. The term 'keyboard interface' is used explicitly, via the definition of 'navigated sequentially', in SC 2.4.3 Focus Order. The term is implicit in other SCs, such as SC 2.4.7 Focus Visible.

(3) Editorial issue in both places:

* "platform device independent services" is awkward. It would be clearer to say "device-independent services of the platform".

Note that issues 1 and 3 were present in the 2013 version of WCAG2ICT ([2.1.1 Keyboard](https://www.w3.org/TR/2013/NOTE-wcag2ict-20130905/#keyboard-operation-keyboard-operable); [glossary term](https://www.w3.org/TR/2013/NOTE-wcag2ict-20130905/#keybrd-interfacedef)). Issue 2 is new.

### Proposed

In context of 2.1.1, change the last sentence to:

> When [software](#software) supports such a device-independent service of the platform, and the software's functionality is made operable through the service, then this success criterion would be satisfied.

In context of the glossary definition of 'keyboard interface', change the last sentence to:

> [Software](https://github.com/w3c/wcag2ict/issues/421#software) may support operation via such device-independent services of the platform.

**Comment by GreggVan** [comment]
almost 
a few fixes

1. you forgot the word  "fully".    Without it - the software could accept keyboard input, but not allow full operation of the software from the keyboard input.

2. it isn't the service, but input from the service... that must control the software

3. this applies to software and non-web documents.  (one could argue that any documents that have functionality are software but we don't do this elsewhere I don't think.  We certainly don't for web pages.)

4. I think it's also useful to mention what this software service is more explicitly.   It gets lost and there are lots of platform services. 

FIXING THESE YOU GET:

> When [software](https://github.com/w3c/wcag2ict/issues/421#software) supports such a device-independent service of the platform **for keyboard input to the software or non-web document**, and the software or non-web document functionality is made **fully** operable through **input from** the service, then this success criterion would be satisfied


AND AS YOU POINT OUT -- THERE IS A NEED FOR CHANGE IN BOTH PLACES  (adding my clarifying phrase

"platform device independent services" is awkward. It would be clearer to say "device-independent services of the platform for keyboard input to the software or non-web documen".

**Comment by maryjom** [comment]
I need to dig back to into what the TF agreed for the text to see if something was missed in the transposition of the content into the draft before we start editing and changing things again.

**Comment by maryjom** [comment]
Found the [survey results](https://www.w3.org/2002/09/wbs/55145/wcag2ict-updated-proposals-24May/results#xq4), [resolution](https://www.w3.org/2024/05/30-wcag2ict-minutes#r04), and [meeting discussion](https://www.w3.org/2024/05/30-wcag2ict-minutes#t06) on the changes we made to this definition. Will need to then make comparison of the language to see if something went awry in the translation. Changes had been incorporated using [PR 378 changes to the definitions](https://github.com/w3c/wcag2ict/pull/378/files#diff-d97c4e69cc6b3fc8133a50416ebf94844c485cec94f05fcc7afe153703b4a636) in lines 341-344.

**Comment by samogami** [comment]
I am troubled that we are re editing this content after we worked and came to consensus on this. 

See resolution from [TF meeting on June 13th](https://www.w3.org/2024/06/13-wcag2ict-minutes#t06)

**Comment by GreggVan** [comment]
@samogami   Consensus allows us to move on - and to accept language as the best we can.   And once we make a decision -- the rule is that that pretty much closes it unless new input comes along  (or VERY strong evidence that we got it wrong).  We do that so we don't keep flip flopping.   But it does not prevent us from considering something new.  Better language or new issue.

**Comment by maryjom** [comment]
@samogami The resolution you found is about 2.1.1 Keyboard content in the SC Problematic for Closed Functionality section. This issue is about the general keyboard guidance in Applying SC 2.1.1 Keyboard to non-web documents and software and the links provided in my Jul. 18 comment points to that decision. There are a couple of issues with the text that we reached consensus on...readability and technical correctness of that note and on the definition the note is talking about satisfying success criterion without being specific about which SC. It's worth trying to address these issues if we can.

**Comment by maryjom** [comment]
The TF reached resolutions on 29 August.
- [29 Aug. resolution for 'keyboard interface' definition](https://www.w3.org/2024/08/29-wcag2ict-minutes.html#r03) to remove the last sentence from Note 1 and add to the parenthetic statement to the definition of 'keyboard interface'.
- [29 Aug.resolution for SC 2.1.1 Keyboard](https://www.w3.org/2024/08/29-wcag2ict-minutes.html#r04) to update Note 1 to change the parenthetic statement in 2.1.1 Keyboard for consistency with the definition, but keeping the last sentence of the note.

See PR #494 which will be merged, once reviewed by the editors.

**Comment by maryjom** [comment]
Closing as completed. Changes have been merged.
